flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/ContinuousFileReaderOperator.java

```diff
@@ -380,10 +380,11 @@ public void processWatermark(Watermark mark) throws Exception {
 	public void dispose() throws Exception {
 		Exception e = null;
 		if (state != ReaderState.CLOSED) {
+			state = ReaderState.CLOSED;
```

Comment:
I guess we could add a test simple test coverage for this `state = ReaderState.CLOSED`?

Reply:
Calling `dispose()` twice or calling `close()` after `dispose()` already violates `StreamOperator` lifecycle. I don't think that reaction to "wrong" lifecycle should be tested.

Reply:
If it's not worth testing, then why do we need this code?

Reply:
Reverted the change.